lib/surface/compiler/parser.ex

```diff
@@ -288,26 +288,16 @@ defmodule Surface.Compiler.Parser do
     state.translator.handle_attribute(name, value, meta, state, context)
   end
 
-  defp translate_attr(state, context, {name, {:string, "true", %{delimiter: nil}}, meta}) do
-    meta = Map.put(meta, :unquoted_string?, true)
-    state.translator.handle_attribute(name, true, meta, state, context)
-  end
-
-  defp translate_attr(state, context, {name, {:string, "false", %{delimiter: nil}}, meta}) do
-    meta = Map.put(meta, :unquoted_string?, true)
-    state.translator.handle_attribute(name, false, meta, state, context)
-  end
-
-  defp translate_attr(state, context, {name, {:string, value, %{delimiter: nil}}, meta}) do
-    meta = Map.put(meta, :unquoted_string?, true)
-
-    case Integer.parse(value) do
-      {int_value, ""} ->
-        state.translator.handle_attribute(name, int_value, meta, state, context)
-
-      _ ->
-        raise parse_error("unexpected value for attribute \"#{name}\"", meta)
-    end
+  defp translate_attr(_state, _context, {name, {:string, value, %{delimiter: nil}}, meta}) do
+    raise parse_error(
+            """
+            Surface does not support passing non-string attribute values as literals \
+            (i.e. `selected=true` or `tabindex=3`).
+
+            Hint: replace `#{name}=#{value}` with `#{name}={#{value}}`
+            """,
+            meta
+          )
   end
```

Comment:
One thing just crossed my mind. 

Since unquoted attribute values are accepted in HTML, maybe we should keep this as a warning and the conversion to string, which will keep the rendering consistent with the original HTML. This way copied/pasted HTML would still run and the user will have the opportunity to see all occurrences of this problem instead of a single error pointing to the first one.

We could also try go give a bit of context about the motivation. Something like:

```
the use of unquoted attribute values is not recommended as they will always be converted to strings.

For intance, `selected=true` and `tabindex=3` will be translated to `selected="true"` and `tabindex="3"` respectively.

Hint: if you want to pass a literal boolean or integer, replace `#{name}=#{value}` with `#{name}={#{value}}`
```

WDYT?

Reply:
that makes sense - I'll make that change